### ISSUE TITLE
[BUG]: ColossalAI/examples/images/diffusion/—The results of the sampling were poor

### ISSUE THREAD

**Issue by LhaoH** [hpcaitech/ColossalAI#2688]
### 🐛 Describe the bug

The results of the sampling were poor
plms sampling：
![图片](https://user-images.githubusercontent.com/50076227/218629890-cb88f7f0-8f12-4312-8550-518f129ddaf7.png)
dpm sampling：
![图片](https://user-images.githubusercontent.com/50076227/218629961-e9907fdd-75de-411b-97f7-69331fc2defc.png)
Train 3000 epochs using the TEYVAT dataset

### Environment

# dpm or plms sampling
python txt2img.py --prompt "Teyvat, Name:Keqing, Element:Electro, Weapon:Sword, Region:Liyue, Model type:Medium Female, Description:an anime character wearing a purple dress and cat ears." \
	--dpm \
    --outdir ./output/fashion \
    --ckpt /home/project/ColossalAI/examples/images/diffusion/logfiles/2023-02-13T01-50-34_train_colossalai_teyvat/diff_tb/version_0/checkpoints/epoch=2917-step=11672.ckpt \
    --config /home/project/ColossalAI/examples/images/diffusion/logfiles/2023-02-13T01-50-34_train_colossalai_teyvat/configs/2023-02-13T01-50-34-project.yaml \
  	--n_samples 2

**Comment by LhaoH** [comment]
I don't know what caused it. I tried many different trainings, PLMS sampling was always bad, DPM was a little better, but still there was noise. Looking forward to your reply! Thank you!

**Comment by JThh** [comment]
I will try this out myself and get back to you within the next two days.

**Comment by FrankLeeeee** [comment]
How many steps have you trained?

**Comment by LhaoH** [comment]
@FrankLeeeee 
I train 3000 epochs using the TEYVAT dataset（https://huggingface.co/datasets/Fazzie/Teyvat）, I just modified the configuration file'max_epochs（train_colossalai_teyvat.yaml).
![图片](https://user-images.githubusercontent.com/50076227/219001486-813c9d3f-908c-4fdc-a2ce-93db1451fbd7.png)

**Comment by LhaoH** [comment]
Based on the last generated ckpt file(epoch=2917-step=11672.ckpt )，steps should be 11672.
At this step I saw that the loss was basically not optimized, so I stopped training at epoch 2917.

**Comment by Thomas2419** [comment]
Is there any update on this? @JThh

**Comment by binmakeswell** [comment]
Hi @LhaoH @Thomas2419 Diffusion convergence is a challenging task, and Colossal-AI provides a more effective development tool. This issue does not seem to be a bug caused by Colossal-AI. For personalized development issues, this goes beyond the scope of open source community support. 

Feel free to contact me via email [ybl@hpcaitech.com](mailto:ybl@hpcaitech.com) to discuss formal collaboration, and you will receive professional high-priority support to help you get your product development done quickly and at low cost. Thanks.

This issue was closed due to inactivity. Thanks.

**Comment by JThh** [comment]
@LhaoH Your procedure is basically for fine-tuning the model which won't make meaningful improvement directly on pretrained checkpoints, due to the limited dataset size. Have you run the training procedure using LAION dataset?

**Comment by LhaoH** [comment]
I did not try it on the LAION dataset, but I ran 2 epochs on the custom dataset (14w), totaling 5054 steps, and the generation effect was relatively poor.
prompt:"with slightly puffed shoulder and a flattering u neckline this cozy long sleeved top is easy to wear and effortlessly feminine"
![图片](https://user-images.githubusercontent.com/50076227/233026500-e817ada6-1037-4d67-8b05-c92aa488ec2e.png)
prompt:"sharp vintage frame define stylish sunglass with indelible appeal"
![图片](https://user-images.githubusercontent.com/50076227/233026832-dde8e5fa-6c18-4bee-8906-35c5acb86ec2.png)

 Because I am unable to solve this problem,So I have given up using colosalai. Currently, but I have successfully run the official fine-tuning code on my own dataset. Thank you for keeping this issue in mind.Thank you very much.
